content/articles/loading-image-with-glide-in-android/index.md

```diff
@@ -0,0 +1,306 @@
+    android:layout_height="wrap_content"
+    android:layout_marginTop="16dp"
+    android:text="Drawable"
+    app:layout_constraintStart_toStartOf="@+id/buttonScaling1"
+    app:layout_constraintTop_toBottomOf="@+id/buttonScaling1" />
+<Button
+    android:id="@+id/buttonPlaceholder"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:layout_marginStart="8dp"
+    android:text="Placeholder"
+    app:layout_constraintBottom_toBottomOf="@+id/buttonDrawable"
+    app:layout_constraintStart_toEndOf="@+id/buttonDrawable"
+    app:layout_constraintTop_toTopOf="@+id/buttonDrawable" />
+<Button
+    android:id="@+id/buttonError"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:layout_marginStart="8dp"
+    android:text="Error"
+    app:layout_constraintBottom_toBottomOf="@+id/buttonCircular"
+    app:layout_constraintStart_toEndOf="@+id/buttonCircular"
+    app:layout_constraintTop_toTopOf="@+id/buttonCircular" />
+<Button
+    android:id="@+id/buttonCache"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:layout_marginTop="16dp"
+    android:text="Cache"
+    app:layout_constraintStart_toStartOf="@+id/buttonDrawable"
+    app:layout_constraintTop_toBottomOf="@+id/buttonDrawable" />
+<Button
+    android:id="@+id/buttonCircular"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"android:layout_marginStart="8dp"
+    android:text="CircleCrop"
+    app:layout_constraintBottom_toBottomOf="@+id/buttonCache"
+    app:layout_constraintStart_toEndOf="@+id/buttonCache"
+    app:layout_constraintTop_toTopOf="@+id/buttonCache" />
+<Button
+    android:id="@+id/buttonTarget"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:layout_marginTop="8dp"
+    android:text="Target"
+    app:layout_constraintStart_toStartOf="@+id/buttonCache"
+    app:layout_constraintTop_toBottomOf="@+id/buttonCache" />
+</androidx.constraintlayout.widget.ConstraintLayout>
+```
+### Step 4: Implementing Glide features
+
+#### Loading an image from URL
+Glide enables the display of images from links (URL) without downloading them.
+Add the following lines of code in the `MainActivity.kt` file:
+
+```kotlin
+binding.buttonUrl.setOnClickListener {
+Glide.with(this)
+    .load(resizeImage)
+    .into(binding.imageView)
+}
+```
+#### Loading Drawable Image
+Glide allows loading of `Drawable` image files to `ImageViews`.
+In the `MainActivity.kt` file, add the following lines of code:
+``` kotlin
+binding.buttonDrawable.setOnClickListener {
+Glide.with(this)
+    .load(R.drawable.image)
+    .into(binding.imageView)
+}
+```
+### Transformations
+In Glide, transformations take a resource, mutate it, and return the mutated resource. They are also used to crop or apply filters to `Bitmaps`.
+
+### a) Center Crop
+Is implemented by using the `centerCrop()` method.
```

Comment:
Give a short description/definition before implementing a concept. Do the same for the other transformations.